### PR TITLE
[release/6.0] Add packaging changes for System.Text.Json

### LINKED ISSUE #61694: [release/6.0] We should enforce that services packages specify `ServicingVersion` when in a servicing release.
See more details here: https://github.com/dotnet/runtime/pull/61688#issuecomment-970677440

cc: @ericstj @Anipik

### PR DESCRIPTION
**Pull request by safern**
It seems like we missed packaging changes for JSON on: https://github.com/dotnet/runtime/commit/fd61aefcf12382a9ddd24295f660b95606201d57

Also fixes: https://github.com/dotnet/runtime/issues/61694

### CODE REVIEW

**Review by Anipik — APPROVED** [review]

**Comment by safern** [comment]
@Anipik @ericstj could you please review: https://github.com/dotnet/runtime/pull/61703/commits/28b425b11e50b92a602a379e37439dabd50f903a

By setting these properties on `packaging.targets` this was happening too late, after `Version.BeforeCommonTargets.targets` was imported which is where the `Version` property is set. So if we want to use arcade version calculation we need to import using the before common targets hooks. The reason why we were seeing package versions `6.0.1` was because the `PatchVersion` is set to 1. 

Also, instead of setting `VersionPrefix` I had to instead set `PatchVersion` to `ServicingVersion` because arcade relies on that and overrides the `VersionPrefix` when these properties are set:
https://github.com/dotnet/arcade/blob/53cc1bc2e555aa7aea95884575d22e21d63708cf/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.BeforeCommonTargets.targets#L76

We could avoid this complexity by just setting `<Version>Major.Minor.ServicingVersion</Version>` and then appending the `VersionSuffix` if not empty, basically copying this logic: https://github.com/dotnet/arcade/blob/53cc1bc2e555aa7aea95884575d22e21d63708cf/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.BeforeCommonTargets.targets#L165

But it felt more right to not duplicate arcade logic on our side.

**Comment by safern** [comment]
After giving it a thought, I decided to go the simple route which is setting `Version` on our side.

File: src/libraries/System.DirectoryServices/Directory.Build.props [hunk 1]
```diff
@@ -5,6 +5,9 @@
          plan on shipping a new desktop version out of band. Instead add API
          to a different assembly. -->
     <AssemblyVersion>4.0.0.0</AssemblyVersion>
+    <!-- Since this assembly version is pinned, we don't want to validate that it matches
+    the expected assembly version on the targeting pack. -->
```
Comment (ericstj) [inline src/libraries/System.DirectoryServices/Directory.Build.props:9]: Is this assembly actually in the targeting pack?  I wouldn't have expected it to be.

Comment (ericstj) [reply src/libraries/System.DirectoryServices/Directory.Build.props:9]: I see, it is in the WindowsDesktop targeting pack.

File: eng/packaging.targets [hunk 2]
```diff
@@ -267,9 +273,15 @@
   </Target>
 
   <Target Name="ValidateAssemblyVersionsInRefPack" 
-          Condition="$(_AssemblyInTargetingPack) == 'true' and '$(PreReleaseVersionLabel)' == 'servicing'" 
+          Condition="'$(SkipValidateAssemblyVersion)' != 'true' and '$(_AssemblyInTargetingPack)' == 'true' and '$(PreReleaseVersionLabel)' == 'servicing'"
           AfterTargets="CoreCompile" >
     <Error Condition="'$(AssemblyVersion)' != '$(LastReleasedStableAssemblyVersion)'" Text="AssemblyVersion should match last released assembly version $(LastReleasedStableAssemblyVersion)" />
```
Comment (ericstj) [inline eng/packaging.targets:278]: Is this all the validation we're doing @Anipik?  I thought we were going to run a reverse check here: https://github.com/dotnet/runtime/blob/c1e9122f0c3202545aee28243c18b0f33c5c12d9/src/libraries/shims/ApiCompat.proj#L103-L108  
To ensure folks don't add API in servicing.

File: src/libraries/System.Text.Json/src/System.Text.Json.csproj [hunk 3]
```diff
@@ -9,6 +9,8 @@
     <Nullable>enable</Nullable>
     <IncludeInternalObsoleteAttribute>true</IncludeInternalObsoleteAttribute>
     <IsPackable>true</IsPackable>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <ServicingVersion>1</ServicingVersion>
```
Comment (ericstj) [inline src/libraries/System.Text.Json/src/System.Text.Json.csproj:13]: Good catch (y)

**Review by ericstj — APPROVED** [review]

**Comment by ericstj** [comment]
Don't treat my comment as blocking.  Merge when ready.

File: eng/packaging.targets (hunk 2 above)
Comment (safern) [reply eng/packaging.targets:278]: We have an open issue, we haven't enabled it.

**Comment by safern** [comment]
Failure is: https://github.com/dotnet/runtime/issues/50748

**Pull request merged**